complex/src/lib.rs

```diff
@@ -763,6 +900,36 @@ impl<T> serde::Deserialize for Complex<T> where
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct ParseComplexError {
+    kind: ComplexErrorKind,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum ComplexErrorKind {
```

Comment:
This is not as straightforward as I thought it would be. I tried to parameterize on `T: FromStr` and ended up having to add a few traits to ComplexErrorKind. I'm still pretty new to Rust, so let me know if these are reasonable or if I'm missing anything.